cmd/thanos/compact.go

```diff
@@ -230,8 +230,12 @@ func runCompact(
 				duplicateBlocksFilter,
 			}, []block.MetadataModifier{block.NewReplicaLabelRemover(logger, conf.dedupReplicaLabels)},
 		)
-		cf.UpdateOnChange(func(blocks []metadata.Meta, err error) {
-			compactorView.Set(blocks, err)
+		cf.UpdateOnChange(func(metas []metadata.Meta, err error) {
+			blocks, blkerr := getBlockMetaAndSize(metas, bkt)
```

Comment:
I think instead of calculating the block size when fetching from the storage It should calculate the block size before uploading it to block storage so that the uploaded block metafile has the size already in in.


@bwplotka what do you think?

Reply:
yeah that seems to be fair, that way we can eliminate a few things and read size from the meta file itself whenever required

Reply:
> I think instead of calculating the block size when fetching from the storage It should calculate the block size before uploading it to block storage so that the uploaded block metafile has the size already in in.

This seems like a nice idea. We will save a lot of network requests. The only thing that we need to think about is how to handle old data. The old blocks won't have any size field in their meta.json.

Reply:
hhm, good question, I think since it is a new feature it is ok to NOT show the size for older blocks

Reply:
@krasi-georgiev, the method you mentioned requires pool which in case is `chunkenc.Pool`  instance and thanos bucket uses `Pool` instance ?

Reply:
you can pass nil